src/maggma/stores/mongolike.py

```diff
@@ -366,15 +366,28 @@ class MongoURIStore(MongoStore):
     client parameters via TXT records
     """
 
-    def __init__(self, uri: str, database: str, collection_name: str, **kwargs):
+    def __init__(self, uri: str, collection_name: str, database: str = None, **kwargs):
         """
         Args:
             uri: MongoDB+SRV URI
             database: database to connect to
             collection_name: The collection name
         """
         self.uri = uri
-        self.database = database
+
+        # parse the dbname from the uri
+        if database is None:
+            with MongoClient(self.uri) as conn:
+                try:
+                    db = conn.get_default_database()
+                except ConfigurationError:
+                    raise ConfigurationError(
+                        "If database name is not supplied, a database must be set in the uri"
+                    )
+            self.database = db.name
+        else:
+            self.database = database
```

Comment:
does `self.database` even need to be stored if it's not provided; IE, does anything break if `database = None`? Also, this should probably be in `connect` ? Connection on init can be bad.

Reply:
Ok using the Mongoclient there was a little hacky.
I have changed it.
Technically the code will not break if `database == None` but it will break if `database == None` and no value is found in the URI.
I do think the `database` should be kept since every Mongo-like store must have a database anyways.